src/createClient.js

```diff
@@ -43,7 +43,7 @@ function createClient (options) {
         auth(client, options)
         break
       case 'microsoft':
-        microsoftAuth.authenticate(client, options)
+        microsoftAuth.authenticate(client, options).catch((err) => client.emit('error', err))
```

Comment:
what about other auth methods? why is it different here ?

Reply:
With the old mojang login the auth function was synchronous. Whenever it had an error it would emit the error event on the bot. The new microsoft authenticate function is async. When it has an error it rejects the promise. This would cause a unhandled promise rejection warning in older node versions. But new node versions treat unhandled promise rejections as process errors and exit the node process. This change proberly handles the promise returned by the authenticate function.